helix-term/src/ui/picker.rs

```diff
@@ -512,6 +512,8 @@ impl<T: 'static + Send + Sync, D: 'static + Send + Sync> Picker<T, D> {
         if self.query == old_query {
             return;
         }
+        // If the query has meaningfully changed, reset the cursor to the top of the results.
+        self.cursor = 0;
```

Comment:
👏 Such a small thing, but it trips me up every once in a while with the old picker. Was about to suggest this change!